octokit/search.go

```diff
@@ -0,0 +1,34 @@
+package octokit
+
+import (
+	"github.com/jingweno/go-sawyer/hypermedia"
+	"net/url"
+)
+
+var SearchURL = Hyperlink("search{/type}")
+
+func (c *Client) UserSearches(url *url.URL) (searches *UserSearchService) {
+	searches = &UserSearchService{client: c, URL: url}
+	return
+}
+
+// A service to return search records
+type UserSearchService struct {
+	client *Client
+	URL    *url.URL
+}
+
+// Get a list of search results based on SearchService#URL
+func (g *UserSearchService) All() (userSearchResults UserSearchResults,
```

Comment:
I think we can get by with one service with four methods. How would you feel about:

```go
type SearchService struct {
	client *Client
	URL    *url.URL
}

// Get a list of search results based on SearchService#URL
func (g *SearchService) UserSearch() (userSearchResults UserSearchResults,
	result *Result) {
	result = g.client.get(g.URL, &userSearchResults)
	return
}
```

We can then add `CodeSearch`, `RepositorySearch`, etc.